codec/src/main/java/io/netty/handler/codec/compression/Lz4FrameEncoder.java

```diff
@@ -233,33 +237,25 @@ private ChannelFuture finishEncode(final ChannelHandlerContext ctx, ChannelPromi
         flushBufferedData(footer);
 
         final int idx = footer.writerIndex();
-        final byte[] dest = footer.array();
-        final int destOff = footer.arrayOffset() + idx;
         footer.setLong(idx, MAGIC_NUMBER);
-        dest[destOff + TOKEN_OFFSET] = (byte) (BLOCK_TYPE_NON_COMPRESSED | compressionLevel);
-        writeIntLE(0, dest, destOff + COMPRESSED_LENGTH_OFFSET);
-        writeIntLE(0, dest, destOff + DECOMPRESSED_LENGTH_OFFSET);
-        writeIntLE(0, dest, destOff + CHECKSUM_OFFSET);
+        footer.setByte(idx + TOKEN_OFFSET, (byte) (BLOCK_TYPE_NON_COMPRESSED | compressionLevel));
+        footer.setIntLE(idx + COMPRESSED_LENGTH_OFFSET, 0);
```

Comment:
0 Little Endian == 0 Big Endian ... should we just remove the `LE` and potentially avoid byte swapping?